src/net/larse/lcms/algorithms/VCT.java

```diff
@@ -0,0 +1,1567 @@
+     * pixels. This is predicated on setting a maximum number of possible years
+     * that we will hold for any pixel. Each pixel's input arrays will actually
+     * determine how much of these arrays get used.
+     */
+    private void allocateArrays() {
+
+      // Integer MAX_YEARS arrays
+      this.mask = new int[MAX_YEARS];
+      this.yearTable = new int[MAX_YEARS];
+      this.qFlag = new int[MAX_YEARS];
+      this.cstSeg = new int[MAX_YEARS];
+      this.cstSegSmooth = new int[MAX_YEARS];
+      this.distFlag = new int[MAX_YEARS];
+      this.distYear = new int[MAX_YEARS];
+      this.distLength = new int[MAX_YEARS];
+      this.regrFlag = new int[MAX_YEARS];
+
+      // Double MAX_YEARS arrays
+      this.distR2 = new double[MAX_YEARS];
+      this.distMagn = new double[MAX_YEARS];
+      this.distMagnB4 = new double[MAX_YEARS];
+      this.distMagnVi = new double[MAX_YEARS];
+      this.distMagnBr = new double[MAX_YEARS];
+      this.regrR2 = new double[MAX_YEARS];
+      this.regrSlope = new double[MAX_YEARS];
+      this.regrRough = new double[MAX_YEARS];
+
+      // Double N_BANDS arrays
+      this.meanForUdBx = new double[N_BANDS];
+      this.sdForUdBx = new double[N_BANDS];
+
+      // Double N_BANDS x MAX_YEARS arrays
+      this.ud = new double[N_BANDS][MAX_YEARS];
+    }
+
+    /**
+     * This is the main function for determining change analysis within VCT.
+     * First, time series information is read from a number of different forest
+     * indexes (known as forest z-scores in the paper, ud scores throughout the
+     * code) and a mask image denoting land cover type and/or image artifacts
+     * (e.g. cloud, shadow). The algorithm consists of two main functions and a
+     * cleanup function:
+     *
+     * 1) interpolationAndIndices - The ud series for all indices is first
+     * filled in for all cloud/shadow pixels from neighboring years.
+     *
+     * 2) analyzeUDist - The filled series are analyzed to find anomalous
+     * deviations from neighboring years and characterized into disturbance and
+     * regrowth segments
+     *
+     * 3) setDisturbanceVariables - Various cleanup and clamping of variable
+     * values as well as determining disturbance duration.
+     *
+     * Note that much of the code relies of comparing series values with known
+     * or derived thresholds of "forestness" and anomalies are detected when
+     * these threshold bounds are exceeded.
+     *
+     * @param ud - List of all UD values across all pertinent bands and indices
+     * (B3, B4, B5, B7, thermal, NDVI, DNBR, R45). There is one List per index
+     * which contains values for all years.
+     *
+     * @param mask - List of mask values for this pixel across all years
+     *
+     * @param years - List of years corresponding to the indices in ud and mask
+     *
+     * @return - Unspecified for now
+     */
+    public VCTOutput getResult(List<List<Double>> ud,
```

Comment:
The first thing you do with these lists is turn them into arrays.  Maybe just pass them as arrays here?